### PR TITLE
Antenna flag

### LINKED ISSUE #51: Antenna present flag not reported

### PR DESCRIPTION
**Pull request by soulcmdc**
Antenna flag reporting, and warnings when the incorrect settings are chosen.
Fixes #51 
/cc @mfine @fnoble @denniszollo @mookerji

### CODE REVIEW

**Comment by mfine** [comment]
If we *really* want to incorporate this into the hearbeat, these settings look a-ok to me.

File: src/board/max2769.c [hunk 1]
```diff
@@ -178,8 +185,26 @@ void max2769_setup(void)
   static struct setting_type antenna_setting;
   int TYPE_ANTENNA = settings_type_register_enum(antenna_enum, &antenna_setting);
   SETTING_NOTIFY("frontend", "antenna_selection", antenna, TYPE_ANTENNA, antenna_changed);
+  max2769_ant_flag_setup();
 }
 
+/* Set up the Antenna Flag GPIO */
+void max2769_ant_flag_setup(void)
+{
+  // Ensure Port C is enabled
```
Comment (fnoble) [inline src/board/max2769.c]: `/* */` style not `//` in piksi_firmware

File: src/board/max2769.c [hunk 2]
```diff
@@ -158,13 +160,18 @@ bool antenna_changed(struct setting *s, const char *val)
       break;
     case PATCH:
       max2769_conf1 |= MAX2769_CONF1_LNAMODE_LNA1;
+      if (ant_status) log_warn("Patch antenna selected, but an external "
```
Comment (fnoble) [inline src/board/max2769.c]:
```
if (ant_status) {
  log_warn("...");
}
```

File: src/board/max2769.c [hunk 3]
```diff
@@ -158,13 +160,18 @@ bool antenna_changed(struct setting *s, const char *val)
...
       break;
     case EXTERNAL:
       max2769_conf1 |= MAX2769_CONF1_LNAMODE_LNA2;
+      if (!ant_status) log_warn("External antenna selected, but no external " 
+                                  "antenna appears to be connected.\n"); 
       break;
     }
     max2769_write(MAX2769_CONF1, max2769_conf1);
     log_info("Antenna changed to: %s\n", val);
+
```
Comment (fnoble) [inline src/board/max2769.c]: Newline not needed

File: src/system_monitor.c [hunk 4]
```diff
@@ -142,9 +143,19 @@ static msg_t system_monitor_thread(void *arg)
...
   systime_t time = chTimeNow();
+  
+  bool ant_status=0;
 
   while (TRUE) {
-    u32 status_flags = SBP_MAJOR_VERSION << 8 | SBP_MINOR_VERSION;
+
+    if (ant_status != max2769_ant_status()) {
+      ant_status = max2769_ant_status();
+      if (ant_status && max2769_ant_setting() == AUTO)
```
Comment (fnoble) [inline src/system_monitor.c]: use `{ }` even for one line statements (its in the style guide)

File: src/system_monitor.c [hunk 5]
```diff
@@ -142,9 +143,19 @@ static msg_t system_monitor_thread(void *arg)
   chRegSetThreadName("system monitor");
 
   systime_t time = chTimeNow();
+  
+  bool ant_status=0;
```
Comment (fnoble) [inline src/system_monitor.c]: ` = 0`

File: src/board/max2769.c [hunk 6]
```diff
@@ -178,8 +185,26 @@ void max2769_setup(void)
   static struct setting_type antenna_setting;
   int TYPE_ANTENNA = settings_type_register_enum(antenna_enum, &antenna_setting);
   SETTING_NOTIFY("frontend", "antenna_selection", antenna, TYPE_ANTENNA, antenna_changed);
+  max2769_ant_flag_setup();
```
Comment (fnoble) [inline src/board/max2769.c]: Could inline?

**Comment by soulcmdc** [comment]
@fnoble Style items should be fixed.

**Comment by fnoble** [comment]
Can you squash into a single commit?

**Comment by fnoble** [comment]
merge at will!

**Pull request merged**